src/survey_questions/convert.py

```diff
@@ -0,0 +1,391 @@
+import re
+
+import matplotlib.pyplot as plt
+import pandas as pd
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.metrics.pairwise import cosine_similarity
+
+# --- 1. Import ---
+plik = "questions.csv"
+df = pd.read_csv(plik, sep=",", encoding="utf-8-sig", quotechar='"')
+
+kolumny = [
+    "(1/3) Podaj jedno z pytań, na które odpowiedzi najczęściej szukasz.",
+    "(2/3) Podaj jedno z pytań, na które odpowiedzi najczęściej szukasz.",
+    "(3/3) Podaj jedno z pytań, na które odpowiedzi najczęściej szukasz.",
+]
+
+pytania = df[kolumny].melt(value_name="pytanie")["pytanie"]
+
+
+# --- 2. Cleaning ---
+def oczysc_tekst(t):
+    t = str(t).lower().strip()
+    t = re.sub(r"[^a-z0-9ąćęłńóśźż\s]", "", t)
+    t = re.sub(r"\s+", " ", t)
+    return t
+
+
+pytania = (
+    pytania.dropna()
+    .astype(str)
+    .map(oczysc_tekst)
+    .drop_duplicates()
+    .reset_index(drop=True)
+)
+
+
+# --- 3. Categories ---
+kategorie_slowa = {
+    "sesja": [
+        "sesja",
+        "egzamin",
+        "egzaminy",
+        "kolokwium",
+        "kolos",
+        "test",
+        "termin egzaminu",
+        "terminy egzaminow",
+        "plan sesji",
+        "termin sesji",
+        "obrona",
+        "zaliczenie",
+        "dzien egzaminu",
+        "termin kolokwium",
+        "terminy kolokwiow",
+        "testy",
+        "sesji",
+    ],
+    "plan": [
+        "plan",
+        "zajecia",
+        "plan zajec",
+        "plan studiow",
+        "godzina",
+        "godziny",
+        "grafik",
+        "rozkład",
+        "sala",
+        "dzien",
+        "dni",
+        "zmiana planu",
+        "zamiana dni",
+        "dzien zamieniony",
+        "plan poniedzialkowy",
+        "plan wtorkowy",
+        "o ktorej mam zajecia",
+        "kiedy mam zajecia",
+        "nastepne zajecia",
+        "kolejne zajecia",
+        "sali",
+        "zajęcia",
+        "grupy ćwiczeniowe",
+        "godzinach",
+        "ćwiczenia",
+        "wykłady",
+    ],
+    "sylabus/obieralne": [
+        "sylabus",
+        "syllabus",
+        "ects",
+        "kierunek",
+        "semestr",
+        "program studiow",
+        "program studiów" "przedmiot",
```

Comment:
Tu chyba się program studiów powtórzył dziwnie.
```suggestion
        "przedmiot",
```

Reply:
Tutaj chodzi o to że jeden jest z ó a drugi z o, przecinek poprawiłam